### PR TITLE
Remove distance_type support from GeoDistance

### LINKED ISSUE #15616: Remove the PLANE and ARC distance types
Elasticsearch has 3 ways to compute geo distances:
 - PLANE,
 - ARC,
 - SLOPPY_ARC.

But this is sometimes confusing to our users, see eg #15606. Do we really need to have 3 options or could we just use SLOPPY_ARC all the time? It seems to be like a good trade-off between accuracy and speed?

### LINKED ISSUE #22914: Deprecate and remove distance_type parameter from geo_distance query DSL
This is a follow up to #19846 for deprecating (5.x) and removing (6.0) the `distance_type` parameter from the `geo_distance` query dsl. The idea is to provide a single implementation that is consistent with how Lucene's `LatLonPoint` computes distances.

### PR DESCRIPTION
**Pull request by nknize**
This PR completely removes all trace of the `distance_type` parameter and supporting `GeoDistance` class(es) that was deprecated in 5.3.

closes #22914 
closes #15616

### CODE REVIEW

**Comment by nknize** [comment]
@rjernst, you said you missed reviewing Geo PRs! :)  While this isn't exactly an interesting Geo PR, do you want to quickly review? I'd like to get this merged and off our list of open PRs.

File: core/src/main/java/org/elasticsearch/common/geo/GeoPoint.java [hunk 1]
```diff
@@ -185,4 +186,12 @@ public static GeoPoint fromGeohash(String geohash) {
     public static GeoPoint fromGeohash(long geohashLong) {
         return new GeoPoint().resetFromGeoHash(geohashLong);
     }
+
+    public double distanceFrom(GeoPoint other) {
+        return GeoUtils.arcDistance(lat, lon, other.lat, other.lon);
+    }
+
+    public double distanceFrom(GeoPoint other, DistanceUnit units) {
```
Comment (rjernst) [inline core/src/main/java/org/elasticsearch/common/geo/GeoPoint.java]: It doesn't look like `DistanceUnit` is available in painless (needs to be whitelisted). However, I'm also not sure why we really need this. Can we have just `distanceFrom(point)` above and leave conversion to units other than meters to the user?

File: core/src/main/java/org/elasticsearch/index/query/GeoDistanceQueryBuilder.java [hunk 2]
```diff
@@ -98,7 +89,6 @@ public GeoDistanceQueryBuilder(StreamInput in) throws IOException {
         distance = in.readDouble();
         validationMethod = GeoValidationMethod.readFromStream(in);
         center = in.readGeoPoint();
-        geoDistance = GeoDistance.readFromStream(in);
```
Comment (rjernst) [inline core/src/main/java/org/elasticsearch/index/query/GeoDistanceQueryBuilder.java]: With maintaining the ability for rolling upgrades from 5.last to 6.0, do we need to maintain bwc logic in here (at least to skip over?)?

File: core/src/main/java/org/elasticsearch/common/geo/GeoPoint.java [hunk 3]
```diff
@@ -185,4 +186,12 @@ public static GeoPoint fromGeohash(String geohash) {
     public static GeoPoint fromGeohash(long geohashLong) {
         return new GeoPoint().resetFromGeoHash(geohashLong);
     }
+
+    public double distanceFrom(GeoPoint other) {
```
Comment (rjernst) [inline core/src/main/java/org/elasticsearch/common/geo/GeoPoint.java]: This should be documented, specifically to note its return value is in meters.

**Review by rjernst — COMMENTED** [review]
I left a couple comments. Also, maybe a followup should be to remove the various distance methods for geo in ScriptDocValues? Geo there is exposed as GeoPoint, so users can call `distanceFrom` directly on the `.value` (against another field's `.value`).

**Comment by rjernst** [comment]
@nknize Do you want to bring this PR up to date and address the comments I left?

**Comment by dakrone** [comment]
ping @nknize can you update this PR?

**Comment by javanna** [comment]
another ping @nknize what's the plan with this PR? There are comments to be addressed it seems but there is no activity in the last 7 months. Does it mean that we can close it?

**Comment by codebrain** [comment]
Documentation for this parameter is still present: https://www.elastic.co/guide/en/elasticsearch/reference/6.x/query-dsl-geo-distance-query.html

**Comment by djptek** [comment]
@nknize I've pushed some new commits to address the preceding conflicts

**Comment by djptek** [comment]
new commits are here: https://github.com/elastic/elasticsearch/commit/952c62893e4bcc61e11cd954d4edfd420807a7ca

**Comment by nknize** [comment]
Thank you @djptek I merged from the [nknize-remove](https://github.com/elastic/elasticsearch/tree/nknize-remove/GeoDistance) feature branch into my [dev branch](https://github.com/nknize/elasticsearch/tree/remove/GeoDistance) so that the PR maintains history. I also fixed the precommit failures but haven't checked the tests. You're added as a collaborator so you should be able to check out my branch by cloning my fork into a working directory:

```
git clone git@github.com:nknize/elasticsearch.git nknize-es
git checkout remove/GeoDistance
```
And continuing to commit to that branch.

**Comment by iverase** [comment]
@nknize is this PR only for master and another PR needs to follow for 7.x to deprecate the  setting?

**Comment by nknize** [comment]
@iverase correct. The [deprecation PR](https://github.com/elastic/elasticsearch/pull/23006) is open/stalled and should also be brought current.

**Comment by djptek** [comment]
WIP updating

**Comment by djptek** [comment]
Hi @nknize @iverase I took a fresh look and saw there are many changes - is this still useful?

**Comment by jimczi** [comment]
This is outdated, hence closing.

**Pull request closed without merging**